zebrad/src/components/tracing.rs

```diff
@@ -152,25 +158,56 @@ impl Config {
     }
 }
 
-impl Default for Config {
-    fn default() -> Self {
-        #[cfg(feature = "progress-bar")]
+/// Returns the default log file path based on `log_file` and `progress_bar`.
+fn default_log_file(
+    log_file: Option<PathBuf>,
+    progress_bar: Option<ProgressConfig>,
+) -> Option<PathBuf> {
+    if let Some(log_file) = log_file {
+        return Some(log_file);
+    }
+
+    if progress_bar.is_some() {
         let default_log_file = dirs::state_dir()
             .or_else(dirs::data_local_dir)
             .map(|dir| dir.join("zebrad.log"));
 
+        return default_log_file;
+    }
+
+    None
+}
+
+impl Default for Config {
+    fn default() -> Self {
+        // TODO: enable progress bars by default once they have been tested
+        let progress_bar = None;
+
         Self {
             use_color: true,
             force_use_color: false,
             filter: None,
             buffer_limit: 128_000,
             endpoint_addr: None,
             flamegraph: None,
-            #[cfg(not(feature = "progress-bar"))]
-            log_file: None,
-            #[cfg(feature = "progress-bar")]
-            log_file: default_log_file,
+            progress_bar,
+            log_file: default_log_file(None, progress_bar),
```

Comment:
We could use a lazy static, but since this code is only run once, it adds more code complexity without much benefit.

I also find option-based logic operations harder to read.